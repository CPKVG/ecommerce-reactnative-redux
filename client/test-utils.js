// import { configureStore } from '@reduxjs/toolkit';
// import React from 'react';
// import { render as rtlRender } from '@testing-library/react-native'
// import { Provider } from 'react-redux';
// import { PersistGate } from 'redux-persist/integration/react';
// import { persistor } from './redux/createStore';
// import { rootReducer } from './redux/rootReducer';

// //custom render

// function render(
//     ui ,{
//         preloadedState,
//         store = configureStore({ reducer: rootReducer, preloadedState}),
//         ...renderOptions
//     } = {}
// ){
//   function Wrapper( {childern} ){
//       return <Provider store = {store}>
//           <PersistGate persistor={persistor}>
//             {childern}
//           </PersistGate>
//       </Provider>
//   }  
//   return rtlRender(ui, {wrapper: Wrapper, ...renderOptions})
// }

// //re-export everything
// export * from '@testing-library/react-native'
// //override render method
// export { render }


import React from "react"
import { render } from '@testing-library/react-native';
import { Provider } from 'react-redux';
// import configureStore from 'redux-mock-store';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { rootReducer } from "./redux/rootReducer";

const INITIAL_STATE = { ...rootReducer }; // Or use value from root reducer
const middlewares = [thunk]
const mockStore = configureMockStore(middlewares)
// const mockStore = configureStore([]);
const store = mockStore(INITIAL_STATE);

const reduxRender = (ui, options) =>
  render(ui, {
    
    wrapper: ({ children }) => <
      Provider store={store}>
          {children}
      </Provider>,
    ...options
  });

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { reduxRender as render, store };