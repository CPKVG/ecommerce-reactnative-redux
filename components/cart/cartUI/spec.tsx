import React from "react"

import { cleanup, fireEvent } from "@testing-library/react-native";
import { NavigationContainer } from "@react-navigation/native";
import { navigationRef } from "../../../RootNavigation";
import CartUI from ".";
import { render } from "../../../test-utils";

afterEach(cleanup)
jest.useFakeTimers()

    describe('tesing cartUI component',()=>{
        const component = (
            <NavigationContainer>
                <CartUI/>
            </NavigationContainer>
        )
        test("btn event changes screen to checkout",()=>{
            if (navigationRef.isReady()) {
            const { getByTestId,findByText } = render(component)
            const button = getByTestId('btn')
            
            fireEvent.press(button)
            const newScreen = findByText('Contact Information')
            expect(newScreen).toBeTruthy    
            }
        })

        //Snapshot test
        

    })
