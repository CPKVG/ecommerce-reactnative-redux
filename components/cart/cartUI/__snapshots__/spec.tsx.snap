// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tesing cartUI component renders the UI component 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text>
      UI btn
    </Text>
  </View>
</View>
`;
